extensions/crypto/lib.rs

```diff
@@ -51,6 +59,28 @@ pub fn op_crypto_get_random_values(
   Ok(())
 }
 
+pub async fn op_crypto_subtle_digest(
+  _state: Rc<RefCell<OpState>>,
+  algorithm_id: i8,
+  data: Option<ZeroCopyBuf>,
+) -> Result<ZeroCopyBuf, AnyError> {
+  let algorithm = match algorithm_id {
+    0 => &digest::SHA1_FOR_LEGACY_USE_ONLY,
+    1 => &digest::SHA256,
+    2 => &digest::SHA384,
+    3 => &digest::SHA512,
+    _ => panic!("Invalid algorithm id"),
+  };
+
+  let input = data.ok_or_else(null_opbuf)?;
+  let output = tokio::task::spawn(async move {
```

Comment:
```suggestion
  let output = tokio::task::spawn_blocking(move || {
```
Plain spawn() with the single-threaded runtime means it's still going to block the event loop.

Reply:
Oops, mindless coping from the original.